### PR TITLE
Frontend updates

### PR DESCRIPTION
**Pull request by crangelsmith**
- [x] Adding dates picker (this is just a placeholder and doesn't do anything yet, it will be updated when the dates get connected to the counterfactual simulation endpoint in #76)
- [x] Including country restriction dates information in front end
- [x] Backend minor update: adding querying options in ModelDateRange view (so we can do the next point) 
- [x] Restricting date period shown in histogram to > 20 Feb 2020 (this will change with issue #50 )
- [x] Placeholder component cards for issues #75  and #74
- [x] Reordering of info panel components (preliminary new ordering, this might change again so I haven't worried too much that it is flexible for screen scaling, etc. )

### CODE REVIEW

**Comment by crangelsmith** [comment]
Updated front end info panel for this PR:
Screenshot 2021-04-20 at 17 38 47

File: frontend/src/components/Histogram.jsx [hunk 1]
```diff
@@ -83,6 +105,26 @@ export default class Histogram extends React.Component {
                 dataKey="weekly_avg_counterfactual"
                 stroke="#ff7300"
               />
+              if (this.state.first_restrictions_date != null){" "}
```
Comment (jemrobinson) [inline frontend/src/components/Histogram.jsx:108]: Can this just be `if (this.state.first_restrictions_date)`? I know that would work in some languages but not sure about Javascript.

File: frontend/src/components/InfoPanel.jsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import Col from "react-bootstrap/Col";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Histogram from "./Histogram";
+import MyDatesPicker from "./MyDatesPicker";
```
Comment (jemrobinson) [inline frontend/src/components/InfoPanel.jsx:8]: Can we change this name to either `DatesPicker` or something like `CounterfactualDatesPicker`?

File: frontend/src/components/MyDatesPicker.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import DatePicker from "react-date-picker";
+import DatePicker from "react-date-picker";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import { ResponsiveContainer } from "recharts";
+
+export default class MyDatesPicker extends React.Component {
+  state = {
+    date: [new Date(), new Date()],
+    dateTwo: [new Date(), new Date()],
```
Comment (jemrobinson) [inline frontend/src/components/MyDatesPicker.jsx:12]: Can we change these variable names to be more descriptive (eg. lockdownDate and firstRestrictionsDate)?

File: frontend/src/components/MyDatesPicker.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  onChange = (date) => this.setState({ date });
+  onChangeTwo = (dateTwo) => this.setState({ dateTwo });
```
Comment (jemrobinson) [inline frontend/src/components/MyDatesPicker.jsx:16]: Same with these function names.

File: frontend/src/tasks/LoadDailyCasesTask.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ class LoadDailyCasesTask {
   #getDailyCovidCases = async (datatype, iso_code) => {
     try {
       const res = await axios.get(
-        `http://localhost:8000/api/cases/${datatype}/daily/normalised/?iso_code=${iso_code}&end_date=2020-06-23`,
+        `http://localhost:8000/api/cases/${datatype}/daily/normalised/?iso_code=${iso_code}&end_date=2020-06-23&start_date=2020-02-20`,
```
Comment (jemrobinson) [inline frontend/src/tasks/LoadDailyCasesTask.js:7]: I have no problem with hardcoding start/end dates but I wonder whether there's a way we can do so in a central config file or similar. Not a fix needed for this PR but something to think about.

File: frontend/src/tasks/LoadRestrictionsDatesTask.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      return res.data;
+    } catch (error) {
+      console.log(error);
+      return [];
+    }
+  };
+
+  getCountryRestrictionDates = async (iso_code) => {
+    const data = await this.#getRestrictionDates(iso_code);
+    return data;
```
Comment (jemrobinson) [inline frontend/src/tasks/LoadRestrictionsDatesTask.js:19]: Can this just be `return await this.#getRestrictionDates(iso_code);`? I'm not sure whether that would work or not.

**Review by jemrobinson — COMMENTED** [review]
A couple of minor comments but it looks fine overall

File: frontend/src/tasks/LoadDailyCasesTask.js (hunk 5 above)
Comment (crangelsmith) [reply frontend/src/tasks/LoadDailyCasesTask.js:7]: Yeah, I agree. I'm working on it in PR #77, where the start and end dates are the ones corresponding to each country "first wave".

File: frontend/src/components/InfoPanel.jsx (hunk 2 above)
Comment (crangelsmith) [reply frontend/src/components/InfoPanel.jsx:8]: Yeah, this is being refactored in PR #77 .

File: frontend/src/components/MyDatesPicker.jsx (hunk 3 above)
Comment (crangelsmith) [reply frontend/src/components/MyDatesPicker.jsx:12]: yes, this is being refactored in PR #77, they might become instances of the InfoPanel class, given that this information inherits to several of the sub components.

File: frontend/src/tasks/LoadRestrictionsDatesTask.js (hunk 6 above)
Comment (crangelsmith) [reply frontend/src/tasks/LoadRestrictionsDatesTask.js:19]: will test this in PR #77

**Comment by crangelsmith** [comment]
All comments are being addressed or refactored out in PR #77, so I think I'll merge this PR for now.

**Pull request merged**